#!/usr/bin/env python

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()
#       Name       Type      Level Description     Default Min   Max
gen.add("canny_threshold", double_t, 0, "Canny Edge Detection Threshold", 220, 0, 255)
gen.add("canny_multiplier", double_t, 0, "Canny Edge Detection Multiplier", 3, 1, 5)
gen.add("binary_threshold", double_t, 0, "Binary threshold ", 62, 0, 100)
gen.add("epsilon_approx_poly", double_t, 0, "epsilon approx poly", 0.03, 0.001, 1)
gen.add("epsilon_finer_poly", double_t, 0, "epsilon finer poly", 0.012, 0.001, 1)
gen.add("offset_in_z", double_t, 0, "pose offset in Z", 0.055, 0.01, 0.2)
gen.add("min_area_depth_image", double_t, 0, "min area to filter out cavities in depth image", 450, 400, 2000)
gen.add("max_area_depth_image", double_t, 0, "max area to filter out cavities in depth image", 12000, 10000, 20000)

exit(gen.generate("mir_cavity_detector", "cavity_finder", "CavityFinder"))
